src/tabs/tab-headers.component.ts

```diff
@@ -281,86 +329,75 @@ export class TabHeaders implements AfterContentInit, OnChanges {
 		if (selected) {
 			return selected;
 		}
-		return {headingIsTemplate: false, heading: ""};
+		return { headingIsTemplate: false, heading: "" };
 	}
 
-	public showTabList() {
-		this.tabListVisible = true;
-		const focusTarget = this.allTabHeaders.find(tab => {
-			const tabContainer = tab.nativeElement.parentElement;
-			return !tabContainer.classList.contains("bx--tabs__nav-item--selected");
-		});
-		focusTarget.nativeElement.focus();
+	/**
+	 * Selects `Tab` 'tab' and moves it into view on the view DOM if it is not already.
+	 */
+	public selectTab(ref: HTMLElement, tab: Tab, tabIndex: number) {
+		if (tab.disabled) {
+			return;
+		}
+
+		this.currentSelectedTab = tabIndex;
+		this.tabs.forEach(_tab => _tab.active = false);
+		tab.active = true;
+		tab.doSelect();
 	}
 
-	public hideTabList() {
-		this.tabListVisible = false;
+	public handleScroll() {
+		this.changeDetectorRef.markForCheck();
 	}
 
-	public onDropdownKeydown(event: KeyboardEvent) {
-		switch (event.key) {
-			case " ":
-			case "Spacebar":
-			case "Enter":
-				event.preventDefault();
-				this.showTabList();
-				break;
-			default:
-				break;
+	public handleOverflowNavClick(direction: number, multiplier = 15) {
+		const tabList = this.headerContainer.nativeElement;
+
+		const { clientWidth, scrollLeft, scrollWidth } = tabList;
+		if (direction === 1 && !scrollLeft) {
+			tabList.scrollLeft += this.OVERFLOW_BUTTON_OFFSET;
+		}
+
+		tabList.scrollLeft += direction * multiplier;
+
+		const leftEdgeReached =
+			direction === -1 && scrollLeft < this.OVERFLOW_BUTTON_OFFSET;
+		const rightEdgeReached =
+			direction === 1 &&
+			scrollLeft + clientWidth >= scrollWidth - this.OVERFLOW_BUTTON_OFFSET;
+		if (leftEdgeReached || rightEdgeReached) {
+			if (leftEdgeReached) {
+				this.rightOverflowNavButton.nativeElement.focus();
+			}
+			if (rightEdgeReached) {
+				this.leftOverflowNavButton.nativeElement.focus();
+			}
 		}
```

Comment:
Is the parent `leftEdgeReached || rightEdgeReached`  necessary here?

Reply:
You know what? Probably not. This logic comes from the reference implementation.